yarn-project/simulator/src/public/fixtures/opcode_spammer.ts

```diff
@@ -0,0 +1,1081 @@
+      limit: MAX_L2_TO_L1_MSGS_PER_TX,
+    },
+  ],
+
+  // SSTORE IS NOT STRICTLY A SIDE-EFFECT LIMITED OPCODE
+  // SSTORE to the same slot repeatedly has no limit - it just overwrites the same slot.
+  // Only writing to unique slots would hit MAX_TOTAL_PUBLIC_DATA_UPDATE_REQUESTS_PER_TX.
+  [Opcode.SSTORE]: [
+    {
+      setup: [
+        { offset: 0, value: new Field(42n) }, // value
+        { offset: 1, value: new Field(0x100n) }, // slot (same slot each iteration)
+        { offset: 2, value: new Uint32(0n) }, // revertSize
+      ],
+      targetInstructions: () => [new SStore(0, 0, 1)],
+      cleanupInstructions: () => [new Revert(0, 2, 0).as(Opcode.REVERT_8, Revert.wireFormat8)], // revert with empty
+    },
+  ],
+
+  // EMITUNENCRYPTEDLOG - two configs: minimal (many small logs) and max-size (one large log)
+  [Opcode.EMITUNENCRYPTEDLOG]: [
+    {
+      label: 'Many empty logs, revert, repeat',
+      setup: [
+        { offset: 0, value: new Uint32(0n) }, // logSize = 0 fields (minimal)
+        { offset: 1, value: new Uint32(0n) }, // revertSize
+      ],
+      targetInstructions: () => [new EmitUnencryptedLog(0, 0, 1)], // logOffset doesn't matter when size is 0
+      cleanupInstructions: () => [new Revert(0, 1, 0).as(Opcode.REVERT_8, Revert.wireFormat8)], // revert with empty
+      // Max logs with 0-field content: floor(4096 / 2) = 2048
+      limit: Math.floor(FLAT_PUBLIC_LOGS_PAYLOAD_LENGTH / PUBLIC_LOG_HEADER_LENGTH),
+    },
+    {
+      label: 'One max size log, revert, repeat',
+      setup: [
+        // logSize = MAX_PUBLIC_LOG_SIZE_IN_FIELDS
+        { offset: 0, value: new Uint32(BigInt(MAX_PUBLIC_LOG_SIZE_IN_FIELDS)) },
+        { offset: 1, value: new Uint32(0n) }, // revertSize
+        // NOTE: We don't initialize the log contents and just let it use default values (Field(0n))
+        // so that we save more gas and bytecode space for the Emit.
+        //// Initialize all log content fields to zero (FIELD type)
+        //...Array.from({ length: MAX_PUBLIC_LOG_SIZE_IN_FIELDS }, (_, i) => ({
+        //  offset: 2 + i,
+        //  value: new Field(0n),
+        //})),
+      ],
+      targetInstructions: () => [new EmitUnencryptedLog(0, 0, 2)], // uses logOffset 2 (uninitialized Field(0))
+      cleanupInstructions: () => [new Revert(0, 1, 0).as(Opcode.REVERT_8, Revert.wireFormat8)], // revert with empty
+      limit: 1, // Only 1 max-size log fits
+    },
+  ],
+
+  // ═══════════════════════════════════════════════════════════════════════════
+  // GADGETS - Use non-trivial inputs to avoid special-case optimizations
+  // ═══════════════════════════════════════════════════════════════════════════
+  [Opcode.POSEIDON2]: [
+    {
+      // Poseidon2 takes 4 field elements as input
+      setup: Array.from({ length: 4 }, (_, i) => ({
+        offset: i,
+        value: new Field(BigInt(0xdeadbeef + i * 0x1111)),
+      })),
+      targetInstructions: () => [new Poseidon2(0, 0, 0)],
```

Comment:
Not sure what `0,0,0` is, but if you are reading from the same place that you are writing in this case it would be cool hehe